### PR TITLE
GH-760: Event reminders

### PR DESCRIPTION
**Pull request by Midnite1212**

### CODE REVIEW

File: ui/src/components/helpers/userInformationHelpers.js [hunk 1]
```diff
@@ -124,30 +137,51 @@ const generatePublishedFormLinks = (forms) => {
...
+                    <Text margin="0px 15px">
+                      {`${item['formName']} ${isPaymentRequired(item)}`}
+                    </Text>
+                    <Spacer />
+                  </GridItem>
+                  <GridItem colSpan={3} display="flex" alignItems="center"
+                  >
+                    {signedUp ? (
+                      <Box>
+                        <Button style={signedUpButton}>Signed Up</Button>
```
Comment (albertparedandan) [inline ui/src/components/helpers/userInformationHelpers.js:158]: can we make this button disabled

File: ui/src/components/userProfile/UserProfileMobile.js [hunk 2]
```diff
@@ -262,7 +293,17 @@ const UserProfileMobile = (props) => {
                     >
                       Available Signup Links:
                     </Text>
-                    {generatePublishedFormLinks(formList)}
+                    {generatePublishedFormLinks(unsignedFormList)}
```
Comment (albertparedandan) [inline ui/src/components/userProfile/UserProfileMobile.js:296]: are you missing the other param here?

File: ui/src/components/userProfile/UserProfileMobile.js [hunk 3]
```diff
@@ -262,7 +293,17 @@ const UserProfileMobile = (props) => {
...
+                    <Text
+                      fontWeight="700"
+                      fontSize="0.95rem"
+                      color="#718096"
+                      mt="2.5"
+                      mb="2.5"
+                    >
+                      Your Signups:
+                    </Text>
+                    {generatePublishedFormLinks(signedUpFormList)}
```
Comment (albertparedandan) [inline ui/src/components/userProfile/UserProfileMobile.js:306]: same here

File: ui/src/components/userProfile/UserProfileDesktop.js [hunk 4]
```diff
@@ -116,12 +119,29 @@ const UserProfileDesktop = (props) => {
...
   const fetchPublishedForms = useCallback(async () => {
+    //get all forms
     const { data, status } = await getLoginOnlyFormsRequest();
-
     if (status === 200) {
-      setFormList([...data]);
+      setFormList([...data])
+
+    //find forms the user have signed up for
+      const tempFormList = data
```
Comment (albertparedandan) [inline ui/src/components/userProfile/UserProfileDesktop.js:128]: i think this needs to be a backend API call. UI shouldn't be doing any of these logics.

**Review by albertparedandan — CHANGES_REQUESTED** [review]

File: ui/src/components/userProfile/UserProfileDesktop.js [hunk 5]
```diff
@@ -223,6 +243,7 @@ const UserProfileDesktop = (props) => {
                   md: {
                     borderBottom: '2px solid #0628A3',
```
Comment (tamarayustian) [inline ui/src/components/userProfile/UserProfileDesktop.js:244]: if you have repeated class like this, can you make it assigned to a variable like your `signedUpButton`? makes it cleaner

Comment (tamarayustian) [reply ui/src/components/userProfile/UserProfileDesktop.js:244]: i know you just added the color, but would be nice to have drive-by fixes and improve on our codebase slowly when u can do it

**Review by tamarayustian — CHANGES_REQUESTED** [review]

File: server/api/controllers/forms/get-unsignedup-form.js [hunk 6]
```diff
@@ -0,0 +1,55 @@
+module.exports = {
+  friendlyName: 'Get signed up form route for public use',
```
Comment (tamarayustian) [inline server/api/controllers/forms/get-unsignedup-form.js:2]: inaccurate name of what the api is doing here

File: server/api/controllers/forms/get-unsignedup-form.js [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+
+  description:
+    'Gets all signed up forms from specified user id, but checks if the form is published',
```
Comment (tamarayustian) [inline server/api/controllers/forms/get-unsignedup-form.js:5]: same comment as above

File: server/api/controllers/forms/get-unsignedup-form.js [hunk 8]
```diff
@@ -0,0 +1,55 @@
...
+        isDeleted: false,
+      });
+
+      const formData = await Form.find({
+        isPublished: true,
+        isDeleted: false,
+      });
+
+      const sub = data.map( s => formData.find( i => i.id === s.formId ))
+      const final = formData.filter((i) => !sub.includes(i))
```
Comment (tamarayustian) [inline server/api/controllers/forms/get-unsignedup-form.js:45]: can you add comment on this line just to be extra clear that this API is different from get-signedup-form.js

File: server/config/routes.js [hunk 9]
```diff
@@ -26,6 +26,8 @@ module.exports.routes = {
 
   // Forms APIs
   'get /api/forms/get-form': 'forms/get-form',
+  'get /api/forms/get-signedup-form': 'forms/get-signedup-form',
+  'get /api/forms/get-unsignedup-form': 'forms/get-unsignedup-form',
```
Comment (tamarayustian) [inline server/config/routes.js:30]: also you need to add these apis in `policies.js` and specify the access rights. should be ['isLoggedIn'] i believe

**Review by tamarayustian — CHANGES_REQUESTED** [review]

**Review by tamarayustian — APPROVED** [review]

**Review by ksuhartono97 — APPROVED** [review]

**Pull request merged**